internal/webserver/wsauthconfig/config.go

```diff
@@ -0,0 +1,46 @@
+package wsauthconfig
+
+type OAuth2Conf struct {
+	ClientID     string   `validate:"required"` // OAuth client id (this app)
+	ClientSecret string   // OAuth2 secret (associated with ClientID, optional)
+	RedirectURL  string   `validate:"required"` // where should the OAuth2 provider return the user to
+	Scopes       []string // list of scopes to ask for from the OAuth2 provider
+}
+
+type OIDCConf struct {
+	IssuerURL string `validate:"url,omitempty"`
+
+	// the ones below are only needed if the OIDC discovery mechanism doesn't work
+	AuthURL     string `validate:"omitempty,url"`
+	TokenURL    string `validate:"omitempty,url"`
+	UserInfoURL string `validate:"omitempty,url"`
+	Algorithms  []string
```

Comment:
maybe oneof validation?

Reply:
it's a fallback, so it only gets used if the IssuerURL doesn't work for some reason or isn't set, so it can be left empty ^^